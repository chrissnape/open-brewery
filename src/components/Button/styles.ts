import { StyleSheet } from 'react-native';
import { borderRadius, primaryColour, padding, fontSize } from '../../utils/styles';

export default StyleSheet.create({
  button: {
    backgroundColor: primaryColour,
    borderRadius,
    padding,
  },
  label: {
    color: '#fff',
    fontSize,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
