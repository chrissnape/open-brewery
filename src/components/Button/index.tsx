import React, { FC } from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import styles from './styles';

type Props = {
  label: string;
  onPress: () => void;
}

const BreweryCardComponent: FC <Props> = ({ label, onPress }): JSX.Element => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.button}>
      <Text style={styles.label}>{label}</Text>
    </View>
  </TouchableWithoutFeedback>
);

export default BreweryCardComponent;
