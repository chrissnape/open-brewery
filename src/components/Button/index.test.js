import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

const props = {
  label: 'View more',
  onPress: () => {}
}

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button { ...props } />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});