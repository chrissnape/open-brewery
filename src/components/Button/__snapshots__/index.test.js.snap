// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#065535",
      "borderRadius": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    View more
  </Text>
</View>
`;
